### ISSUE TITLE
Shape-inference does not work for more than 16 operations

### ISSUE THREAD

**Issue by tungld** [onnx/onnx-mlir#10]
shape-inference works well if we have less than or equal to 16 operations, but it does not work for more than 16 operations.

E.g. Running `onnf-opt --shape-inference test.mlir` with the following `test.mlir`:
```mlir
module {                                                                                                                                                                
  func @main_graph(%arg0: tensor<5x5x1x32xf32>) -> tensor<*xf32> {
    %0 = "onnx.Identity"(%arg0) : (tensor<5x5x1x32xf32>) -> tensor<*xf32>
    %1 = "onnx.Identity"(%0) : (tensor<*xf32>) -> tensor<*xf32>
    %2 = "onnx.Identity"(%1) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                   
    %3 = "onnx.Identity"(%2) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                 
    %4 = "onnx.Identity"(%3) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                 
    %5 = "onnx.Identity"(%4) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                 
    %6 = "onnx.Identity"(%5) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                 
    %7 = "onnx.Identity"(%6) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                 
    %8 = "onnx.Identity"(%7) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                 
    %9 = "onnx.Identity"(%8) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                 
    %10 = "onnx.Identity"(%9) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                
    %11 = "onnx.Identity"(%10) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                               
    %12 = "onnx.Identity"(%11) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                               
    %13 = "onnx.Identity"(%12) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                               
    %14 = "onnx.Identity"(%13) : (tensor<*xf32>) -> tensor<*xf32>
    %15 = "onnx.Identity"(%14) : (tensor<*xf32>) -> tensor<*xf32>
    return %15 : tensor<*xf32>
  }
}
```
would result a correct output:
```mlir
module {
  func @main_graph(%arg0: tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32> {
    %0 = "onnx.Identity"(%arg0) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %1 = "onnx.Identity"(%0) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %2 = "onnx.Identity"(%1) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %3 = "onnx.Identity"(%2) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %4 = "onnx.Identity"(%3) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %5 = "onnx.Identity"(%4) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %6 = "onnx.Identity"(%5) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %7 = "onnx.Identity"(%6) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %8 = "onnx.Identity"(%7) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %9 = "onnx.Identity"(%8) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %10 = "onnx.Identity"(%9) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %11 = "onnx.Identity"(%10) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %12 = "onnx.Identity"(%11) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %13 = "onnx.Identity"(%12) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %14 = "onnx.Identity"(%13) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %15 = "onnx.Identity"(%14) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    return %15 : tensor<5x5x1x32xf32>
  }
}
```

If we add one more op into `test.mlir` as follows.
```mlir
module {                                                                                                                                                                
  func @main_graph(%arg0: tensor<5x5x1x32xf32>) -> tensor<*xf32> {
    %0 = "onnx.Identity"(%arg0) : (tensor<5x5x1x32xf32>) -> tensor<*xf32>
    %1 = "onnx.Identity"(%0) : (tensor<*xf32>) -> tensor<*xf32>
    %2 = "onnx.Identity"(%1) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                   
    %3 = "onnx.Identity"(%2) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                 
    %4 = "onnx.Identity"(%3) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                 
    %5 = "onnx.Identity"(%4) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                 
    %6 = "onnx.Identity"(%5) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                 
    %7 = "onnx.Identity"(%6) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                 
    %8 = "onnx.Identity"(%7) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                 
    %9 = "onnx.Identity"(%8) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                 
    %10 = "onnx.Identity"(%9) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                                
    %11 = "onnx.Identity"(%10) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                               
    %12 = "onnx.Identity"(%11) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                               
    %13 = "onnx.Identity"(%12) : (tensor<*xf32>) -> tensor<*xf32>                                                                                                               
    %14 = "onnx.Identity"(%13) : (tensor<*xf32>) -> tensor<*xf32>
    %15 = "onnx.Identity"(%14) : (tensor<*xf32>) -> tensor<*xf32>
    %16 = "onnx.Identity"(%15) : (tensor<*xf32>) -> tensor<*xf32>
    return %16 : tensor<*xf32>
  }
}
```
The output of running `onnf-opt --shape-inference test.mlir` is:
```mlir
module {
  func @main_graph(%arg0: tensor<5x5x1x32xf32>) -> tensor<*xf32> {
    %0 = "onnx.Identity"(%arg0) : (tensor<5x5x1x32xf32>) -> tensor<5x5x1x32xf32>
    %1 = "onnx.Identity"(%0) : (tensor<5x5x1x32xf32>) -> tensor<*xf32>
    %2 = "onnx.Identity"(%1) : (tensor<*xf32>) -> tensor<*xf32>
    %3 = "onnx.Identity"(%2) : (tensor<*xf32>) -> tensor<*xf32>
    %4 = "onnx.Identity"(%3) : (tensor<*xf32>) -> tensor<*xf32>
    %5 = "onnx.Identity"(%4) : (tensor<*xf32>) -> tensor<*xf32>
    %6 = "onnx.Identity"(%5) : (tensor<*xf32>) -> tensor<*xf32>
    %7 = "onnx.Identity"(%6) : (tensor<*xf32>) -> tensor<*xf32>
    %8 = "onnx.Identity"(%7) : (tensor<*xf32>) -> tensor<*xf32>
    %9 = "onnx.Identity"(%8) : (tensor<*xf32>) -> tensor<*xf32>
    %10 = "onnx.Identity"(%9) : (tensor<*xf32>) -> tensor<*xf32>
    %11 = "onnx.Identity"(%10) : (tensor<*xf32>) -> tensor<*xf32>
    %12 = "onnx.Identity"(%11) : (tensor<*xf32>) -> tensor<*xf32>
    %13 = "onnx.Identity"(%12) : (tensor<*xf32>) -> tensor<*xf32>
    %14 = "onnx.Identity"(%13) : (tensor<*xf32>) -> tensor<*xf32>
    %15 = "onnx.Identity"(%14) : (tensor<*xf32>) -> tensor<*xf32>
    %16 = "onnx.Identity"(%15) : (tensor<*xf32>) -> tensor<*xf32>
    return %16 : tensor<*xf32>
  }
}
```

**Comment by doru1004** [comment]
Solved by https://github.com/onnx/onnx-mlir/pull/12

**Comment by tungld** [comment]
Thank you for looking at this issue! We can use “closes #10” in the description box of the PR, so that this issue will be closed automatically when the PR lands.

**Comment by doru1004** [comment]
Resolved.
